### ISSUE TITLE
postcss-import hanging

### ISSUE THREAD

**Issue by jasonreiche** [postcss/postcss-import#392]
My designer made a set of CSS for our courses.  She made a course specific CSS with variables and a master style sheet with all the actual style rules.  It works great in Chrome, Firefox, etc, but of course doesn't work in IE11.  I thought, no big deal, I'll just set up a PostCSS process for making final CSS for cross browser support.  The problem is, every time I try, no matter how I try, the files get selected and hang up in postcss-import.  Please help.  I get no warnings or errors, just a hung screen.  I first tried this using postcss-cli and when this started happening I wrote up a little app.  I've tried putting all the CSS in a flat folder structure and that still has the same results.  I just want the 2 css files inlined so I can then run other plugins across the bundled file. 
 For the life of me I can't figure out what is happening.  No additional bundler or task running in use.

**main.js**
```
// General modules
const fs = require('fs');
const watch = require('node-watch');
// PostCSS Specific modules
const postcss = require('postcss');
const atImport = require('postcss-import');

// Literals
const root = '\\\\internal\\data\\Internet Content\\secure\\CourseResources';
const devFolder = root + '\\css-dev';
const prodFolder = root + '\\css-test';

console.log("CSS Watcher has been started.");

watch(devFolder, { recursive: true }, function(event, devPath) {
    // Get file and partial directory path
    var partDir = devPath.replace(devFolder,"").split('\\');
    var partFile = partDir.pop();
    partDir = partDir.join('\\');
    var prodFile = prodFolder + partDir + '\\' + partFile;
    //
    console.log('%s\\%s changed.', partDir, partFile);
    fs.readFile(devPath, "utf8", (error, css) => {
        postcss()
            .use(atImport({
                root: devFolder
            }))
            .process(css, {
                from: devPath,
                to: prodFile,
                map: { inline: false },
                // parser: safeParser
             })
            .then((out) => {
                writeOutput(out);
            })
            .catch(error => {
                console.error(error);
            });
    });
});


function writeOutput(result) {
    console.log(result);
}
```
**CRW273-theme.css**
```
@import "CRW273.css";
@import "../../../default.css";
```
**CRW273.css**
```
:root {
	--primary-color-dark: #6D4186;
	/**----Make sure it has enough contrast with primary-color-light ----**/
	--primary-color-light: #f4f095;
	--primary-color-lighter: #FFF7E6;
	--alt-color-1: #B61F69;
	/**---- alternative darker bg, can be used for accents and links----**/
	--alt-color-2: #BBE3FC;
	/**---- alternative lighter bg ----**/
	--obj-bg-color: #FFF7E6;
	/**---- this is background for objectives, use one of the light bg colors, text color is --header-text-color ----**/
	--header-bg-color: #FFF7E6;
	/**---- top part of the banner ----**/
	--header-text-color: #401759;
	--heading-bg-color: #401759;
	/**---- bottom part of the banner ----**/
	--heading-text-color: #fff;
	--font-stack: 'Arvo', serif;
	--font-stack-theme: 'roboto', sans-serif;
	--tiny-shadow: 0 1px 2px #d1d1d1;



	/* headings scale*/
	--font-size-1: 1rem;
	--font-size-2: 1.2rem;
	--font-size-3: 1.44rem;
	--font-size-4: 1.728rem;
	--font-size-5: 2.074rem;
	--font-size-6: 2.488rem;
	/* svg icons scaling*/
	--icon-width: 60px;
}

@media (max-width: 800px) {
	:root {
		--font-size-1: 1rem;
		--font-size-2: 1.1rem;
		--font-size-3: 1.2rem;
		--font-size-4: 1.4rem;
		--font-size-5: 1.5rem;
		--font-size-6: 1.7rem;
		/* svg icons scaling*/
		--icon-width: 40px;
	}
}
```
**Excerpt from default.css**
```
table {
    border-bottom: 5px solid var(--primary-color-dark, #182E46);
    background: #f7f4f4;
}
table tbody tr:nth-child(even) td {
    background-color: var(--primary-color-lighter, #AFE2FA);
}
table tbody tr.highlighted td {
    background-color: var(--alt-color-2, #F7B73D);
}
table thead th, table th {
    background-color: var(--primary-color-dark, #182E46);
    color: white;
```

**Comment by RyanZim** [comment]
You have no error handling in your code; you may be missing an error that way. Add a `catch` block to the postcss call.

**Comment by jasonreiche** [comment]
What I pasted above is reduced to just the lines required to reproduce the issue.  I have a .catch and neither .catch or .then fire if postcss-import is assigned to postcss.  On the CLI with Verbose, it will list all the files with @import and them just hang.

**Comment by jasonreiche** [comment]
I figured this out, kindof.  The resolve module was getting stuck in a perpetual loop trying to find a package.json from the folder with the css.  My code and css were on different volumes.  I played with a bunch of configurations, but ultimately the best solution was to remove the from: line as that seemed to be what was pointing resolve at the wrong folder for the package.json.  Now everything is purring like a kitten.  

Seems the issue hits at the ternary in index.js when in this on instance of trying to resolve package.json, the ternary has the correct value on the wrong side

This isn't the best solution as the maps generated are limited, but it's a step forward.
